component/common/loki/wal/config.go

```diff
@@ -12,6 +12,7 @@ const (
 var DefaultWatchConfig = WatchConfig{
 	MinReadFrequency: time.Millisecond * 250,
 	MaxReadFrequency: time.Second,
+	DrainTimeout:     time.Second * 30,
```

Comment:
Is this based on the default grace termination timer?

Reply:
So when stopping a WAL enabled `loki.write`, there's two graceful terminations occurring:
1. first the watchers, which assuming we have a single endpoint config, it will be `Watcher.DrainTimeout`
2. then the clients, which assuming we have a single endpoint config, it will be `QueueConfig.DrainTimeout`

That said, the worst total wait is `Watcher.DrainTimeout + QueueConfig.DrainTimeout`. Since the default graceful termination [is 30 seconds](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/) as you mentioned, let's default to 15 s each so the total sum is 30s.